Misleading error messages relating to spyOn modules. - "Cannot spy on export  "[X]". Module namespace is not configurable in ESM"

### Describe the bug

If I write some code like this: 

```
// main.test.ts
import { vi } from "vitest";
import * as lodash from "lodash";
vi.spyOn(lodash, "after");


```

Then this code will error with 

```typescript
TypeError: Cannot spy on export "after". Module namespace is not configurable in ESM. See: https://vitest.dev/guide/browser/#limitations
 ❯ src/main.test.ts:3:4
      1| import { vi } from "vitest";
      2| import * as lodash from "lodash";
      3| vi.spyOn(lodash, "after");
       |    ^
      4| 

Caused by: TypeError: Cannot redefine property: after
```


This links us to this documentation: 

https://vitest.dev/guide/browser/#limitations

This documentation appears to relate to browser mode specifically. 

However this error occurs whether we are using browser mode or not.* 
This causes some confusion when encountering this error. 

Furthermore - this kind of code _is_ valid in vitest v3. This breaking change is not reflected in in the migration guide. 


https://vitest.dev/guide/migration.html#vitest-4

If I'm understanding it right vitest v4 changes the way spys are applied, and directly use of spyon against external modules is no longer supported at all. 

*Note that we can reproduce a similar error that _does_ only occur in browser mode. 

```typescript
import { vi, describe, it, expect } from "vitest";
import * as foo from "./main";
vi.spyOn(foo, "foo");
```


### Reproduction

https://github.com/dwjohnston/vitest-spyon-module-issue

### System Info

```shell
System:
    OS: macOS 15.6.1
    CPU: (12) arm64 Apple M4 Pro
    Memory: 328.81 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.21.1 - /Users/davidjohnston/.nvm/versions/node/v22.21.1/bin/node
    npm: 11.6.2 - /Users/davidjohnston/.nvm/versions/node/v22.21.1/bin/npm
    pnpm: 10.16.1 - /Users/davidjohnston/Library/pnpm/pnpm
    bun: 1.2.21 - /Users/davidjohnston/.bun/bin/bun
  Browsers:
    Brave Browser: 143.1.85.120
    Chrome: 144.0.7559.59
    Edge: 143.0.3650.139
    Safari: 18.6
  npmPackages:
    @vitest/browser-playwright: ^4.0.17 => 4.0.17 
    vitest: 4.0.17 => 4.0.17
```

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vitest.dev/guide/).
- [x] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/spy/src/index.ts
+++ b/packages/spy/src/index.ts
@@ -410,7 +410,7 @@ export function spyOn<T extends object, K extends keyof any>(
         || error.message.includes('can\'t redefine non-configurable property'))
     ) {
       throw new TypeError(
-        `Cannot spy on export "${String(key)}". Module namespace is not configurable in ESM. See: https://vitest.dev/guide/browser/#limitations`,
+        `Cannot spy on export "${String(key)}". Module namespace is not configurable in ESM. See: https://vitest.dev/guide/mocking/modules#mocking-a-module`,
         { cause: error },
       )
     }

--- a/test/browser/test/mocking.test.ts
+++ b/test/browser/test/mocking.test.ts
@@ -12,7 +12,7 @@ it('spying on an esm module prints an error', () => {
     }
   })()
   expect(error.name).toBe('TypeError')
-  expect(error.message).toMatchInlineSnapshot(`"Cannot spy on export "calculator". Module namespace is not configurable in ESM. See: https://vitest.dev/guide/browser/#limitations"`)
+  expect(error.message).toMatchInlineSnapshot(`"Cannot spy on export "calculator". Module namespace is not configurable in ESM. See: https://vitest.dev/guide/mocking/modules#mocking-a-module"`)
 
   expect(error.cause).toBeInstanceOf(TypeError)
 })

--- a/test/unit/test/mocking/vi-spyOn.test.ts
+++ b/test/unit/test/mocking/vi-spyOn.test.ts
@@ -19,6 +19,39 @@ describe('vi.spyOn() edge cases', () => {
     expect(fn3.length).toBe(3)
   })
 
+  test('spying on a non-configurable ESM namespace export points to the module mocking docs, not browser mode', () => {
+    // Fake an ES module namespace: `Symbol.toStringTag` is 'Module' and the
+    // export is non-configurable, so `Object.defineProperty` throws
+    // "Cannot redefine property" — the same guard the real ESM path hits.
+    const namespace: Record<string, unknown> = {}
+    Object.defineProperty(namespace, Symbol.toStringTag, { value: 'Module' })
+    Object.defineProperty(namespace, 'answer', {
+      value: () => 42,
+      enumerable: true,
+      configurable: false,
+      writable: false,
+    })
+    Object.preventExtensions(namespace)
+
+    const error = (() => {
+      try {
+        vi.spyOn(namespace as any, 'answer')
+        expect.unreachable()
+      }
+      catch (err) {
+        return err as Error
+      }
+    })()
+
+    expect(error).toBeInstanceOf(TypeError)
+    expect(error.message).toContain('Cannot spy on export "answer"')
+    // This error fires in Node too, so it must not send users to the
+    // browser-mode-only docs (issue #9467). It should point at the
+    // environment-agnostic module mocking guide instead.
+    expect(error.message).toContain('https://vitest.dev/guide/mocking/modules#mocking-a-module')
+    expect(error.message).not.toContain('/guide/browser')
+  })
+
   test('can spy on a proxy with undefined descriptor\'s value', () => {
     const obj = new Proxy<{ fn: () => number }>({} as any, {
       get(_, prop) {

